### PR TITLE
Alt-F Crash fix 

### PR DESCRIPTION
**Pull request by daschuer**
This PR does not use the probably broken broken QMenuBar::setNativeMenuBar(false) call. 

Even though I still cannot reproduce the issue. It is obvious from the backtrace in #11320 that there is an issue that happens after 
deleting the platformMenuBar here: https://github.com/qt/qtbase/blob/8c47a86eecd286dbe66ca55ed4deaaf0f901d13b/src/widgets/widgets/qmenubar.cpp#L1882 

I assume that the  final crash happens than here: 
https://github.com/qt/qtbase/blob/8c47a86eecd286dbe66ca55ed4deaaf0f901d13b/src/widgets/widgets/qmenubar.cpp#L1767
because of a dangling pointer in the sub-menu actions.

It is still a puzzling why the crash does not happen here. At least all is still working tested on Linux Mint Mate, Ubuntu Unity and Ubuntu Desktop. 

I will now single step through the code to discover the issue.  

@ronso0 I hope this PR gives us the freedom to consider the best solution in your pending work without the fear of a crash. 
Can you confirm the crash is fixed?

### CODE REVIEW

**Comment by daschuer** [comment]
Debugging does not reveal anything. It turns out hat the stored pointer in the actions is a QPointer that cannot become dangling. 
https://github.com/qt/qtbase/blob/de8538966f51cb1589bfc4032be99b266707e26c/src/widgets/widgets/qmenu_p.h#L462
Also Valgrind is not complaining...
I am really curios if this PR fixes the crash.

File: src/mixxx.cpp [hunk 1]
```diff
@@ -1438,14 +1440,13 @@ void MixxxMainWindow::slotViewFullScreen(bool toggle) {
         // Fix for "No menu bar with ubuntu unity in full screen mode" Bug
         // #885890 and Bug #1076789. Before touching anything here, please read
         // those bugs.
+        QApplication::setAttribute(Qt::AA_DontUseNativeMenuBar, true);
```
Comment (ronso0) [inline src/mixxx.cpp:1443]: In my tests I did exactly that (just at the beginning of `createMenuBar()`, not directly before the call) and I did set it in main.cpp, not in mixxx.cpp.

So I wonder how that makes the difference...
Anyway, crash fixed!

**Comment by ronso0** [comment]
Great, that does the trick :tada: 

Crash fixed on Ubuntu 20.04.5 with xfce + vala-appmenu-plugin
and VM Ubuntu 20.04 with Unity desktop + global menu

**Comment by daschuer** [comment]
Cool, so we can merge this? 
There remains just the issue that we have no explanation for this. 
Maybe a race condition with QPointer?

**Comment by ronso0** [comment]
> @ronso0 I hope this PR gives us the freedom to consider the best solution in your pending work without the fear of a crash.

This can easily be combined with #11313 (we can remove the hotkey popup) which fixes the issue with identical fullscreen shortcuts in OS and Mixxx.

Thank you for looking into this!

**Comment by ronso0** [comment]
FWIW still hitting the debug assert on shutdown because of leaked controls.

**Comment by daschuer** [comment]
Is this related to this PR? Which one is leaked?

**Comment by ronso0** [comment]
Unrelated, this happens every time after recreating the menubar.

File: src/mixxx.cpp [hunk 2]
```diff
@@ -217,6 +217,15 @@ MixxxMainWindow::MixxxMainWindow(QApplication* pApp, const CmdlineArgs& args)
...
 
+#ifdef __LINUX__
+    // If the desktop features a global menubar and we go fullscreen during
+    // startup, move the menubar to the window like in slotViewFullScreen()
+    bool fullscreenPref = m_pSettingsManager->settings()->getValue<bool>(
+            ConfigKey("[Config]", "StartInFullscreen"));
+    QApplication::setAttribute(
+            Qt::AA_DontUseNativeMenuBar,
+            args.getStartInFullscreen() || fullscreenPref);
+#endif // __LINUX__
```
Comment (ronso0) [inline src/mixxx.cpp:228]: Note for future improvements:
we can test for this attribute _before_ setting it, then all recreating and reconnecting can be limited to platforms that actually require/support it.

**Comment by ronso0** [comment]
> Gonna merge this now.

I'm about to add/refine the related comments.

**Comment by ronso0** [comment]
> I'm about to add/refine the related comments.

634cb4421c5

**Comment by daschuer** [comment]
Thank you. I have cherry picked the linked commit.

**Pull request merged**